app/backend/app.py

```diff
@@ -36,7 +36,7 @@
 from approaches.chatreadretrievereadvision import ChatReadRetrieveReadVisionApproach
 from approaches.retrievethenread import RetrieveThenReadApproach
 from approaches.retrievethenreadvision import RetrieveThenReadVisionApproach
-from core.authentication import AuthenticationHelper
+from core.authentication import AuthenticationHelper, AuthError
```

Comment:
Love the decorators! One overall comment: what do you think of moving them into their own file, route_decorators.py or some such? That would keep this file a more manageable size.